### PR TITLE
update api docs to new structure

### PR DESCRIPTION
**Pull request by mashehu**

### CODE REVIEW

File: docs/api/make_lint_md.py [hunk 1]
```diff
@@ -1,53 +1,58 @@
-#!/usr/bin/env python
-
-import fnmatch
-import os
+from pathlib import Path
 
+#!/usr/bin/env python
```
Comment (mirpedrol) [inline docs/api/make_lint_md.py:3]: The shebang should go to the top

File: docs/api/make_lint_md.py [hunk 2]
```diff
@@ -1,53 +1,58 @@
...
+import nf_core.commands_pipelines
```
Comment (mirpedrol) [inline docs/api/make_lint_md.py:4]: do we need to import `nf_core.commands_pipelines`?

File: docs/api/_src/api/pipelines/index.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# API Reference
```
Comment (mirpedrol) [inline docs/api/_src/api/pipelines/index.md:1]: Shouldn't this title be `pipeline lint test`? As it's written [here](https://github.com/nf-core/tools/pull/3054/files#diff-4771afee2d8066fa3be4ba4965a28cd8afd54a623054e758a5f5182fd1ed0405R48)

**Review by mirpedrol — APPROVED** [review]

File: docs/api/_src/api/pipelines/index.md (hunk 3 above)
Comment (mashehu) [reply docs/api/_src/api/pipelines/index.md:1]: for "historical" reasons we have all pipeline commands there as an API reference https://nf-co.re/docs/nf-core-tools/api_reference/dev/api/bump_version. not sure if we want to keep them or add the modules and subworkflow commands as well...

Comment (mirpedrol) [reply docs/api/_src/api/pipelines/index.md:1]: ah ok I was confused between linting tests and commands, let's leave it like it is for now

**Pull request merged**